lib/Doctrine/DBAL/Connection.php

```diff
@@ -425,9 +425,46 @@ private function getDatabasePlatformVersion()
 
         // If not connected, we need to connect now to determine the platform version.
         if (null === $this->_conn) {
-            $this->connect();
+            try {
+                $this->connect();
+            } catch (\Exception $originalException) {
+                if (empty($this->_params['dbname'])) {
```

Comment:
No I can't because it is possible to instantiate a connection object without this parameter. I even fell into this trap during writing the tests.